paymentapi/src/main/java/fi/hel/verkkokauppa/payment/api/PaytrailPaymentController.java

```diff
@@ -94,4 +108,52 @@ public ResponseEntity<String> getPaymentUrl(
         }
     }
 
+    @GetMapping("/subscription/get/card-form-parameters")
+    public ResponseEntity<TreeMap<String, String>> getCardFormParameters(
+            @RequestParam(value = "merchantId") String merchantId,
+            @RequestParam(value = "namespace") String namespace
+    ) {
+        try {
+            String paytrailMerchantId = commonServiceConfigurationClient.getMerchantConfigurationValue(merchantId, namespace, ServiceConfigurationKeys.MERCHANT_PAYTRAIL_MERCHANT_ID);
+
+            if (paytrailMerchantId == null) {
+                throw new CommonApiException(
+                    HttpStatus.INTERNAL_SERVER_ERROR,
+                    new Error("failed-to-get-paytrail-merchant-id", "failed to get paytrail merchant id, merchantId: " + merchantId + ", namespace: " + namespace)
+                );
+            }
+
+            String secretKey = commonServiceConfigurationClient.getMerchantPaytrailSecretKey(merchantId);
+
+            if (secretKey == null) {
+                throw new CommonApiException(
+                    HttpStatus.INTERNAL_SERVER_ERROR,
+                    new Error("failed-to-get-merchant-paytrail-secret-key", "failed to get paytrail secret key, merchantId: " + merchantId)
+                );
+            }
+
+
+            String paymentExperienceUrl = experienceUrls.getPaymentExperienceUrl();
+
+            TreeMap<String, String> parameters = new TreeMap<>();
+            parameters.put("checkout-account", paytrailMerchantId);
+            parameters.put("checkout-algorithm", CheckoutAlgorithm.SHA256.toString());
+            parameters.put("checkout-method", CheckoutMethod.POST.toString());
+            parameters.put("checkout-nonce", UUIDGenerator.generateType4UUID().toString());
+            parameters.put("checkout-timestamp", Instant.now().toString());
+            parameters.put("checkout-redirect-success-url", String.format("%spaytrailCard/redirect/success", paymentExperienceUrl));
+            parameters.put("checkout-redirect-cancel-url", String.format("%spaytrailCard/redirect/cancel", paymentExperienceUrl));
```

Comment:
We should not append paths to urls, because we need to test these callbacks locally. This whole data/path should come from one environment variable like other callbacks.